src/world.rs

```diff
@@ -0,0 +1,72 @@
+mod world_generator;
+pub mod block_textures;
+pub mod block_data;
+
+use crate::Renderer;
+use crate::Player;
+
+pub struct World {
+    pub world_map: Box<[u16; (Self::MAP_WIDTH * Self::MAP_HEIGHT) as usize]>,
+}
+
+impl World {
+    pub const MAP_WIDTH: u32 = 4000;
+    pub const MAP_HEIGHT: u32 = 500;
+
+    pub fn update_frame_mesh(&self, player: &Player, mesh: &mut egui::Mesh, window_width: f32, window_height: f32) {
+        let texture_pixel_width: f32 = Renderer::BLOCK_PIXEL_COUNT / 5.0;
+                
+        let screen_block_count_x: f32 = window_width / Renderer::BLOCK_PIXEL_COUNT;
+        let screen_block_count_y: f32 = window_height / Renderer::BLOCK_PIXEL_COUNT;
+
+        let player_block_x: i32 = (player.position_x / Renderer::BLOCK_PIXEL_COUNT).trunc() as i32;
+        let player_block_y: i32 = (player.position_y / Renderer::BLOCK_PIXEL_COUNT).trunc() as i32;
+
+        for relative_x in (-screen_block_count_x / 2.0).floor() as i32..(screen_block_count_x / 2.0).ceil() as i32 + 1 {
+            for relative_y in (-screen_block_count_y / 2.0).floor() as i32..(screen_block_count_y / 2.0).ceil() as i32 + 1 {
+                for texture_x in 0..5 {
```

Comment:
I think this is the right call, since your block textures are nice and simple, but if the inner loops grew a bit more I would think about having some kind of render block method to split up world <=> block <=> pixel